solidusio/solidus
# Use SolidusPaypalCommercePlatform >= 1.a if frontend is StarterFrontend

## Summary

Fixes #4788.

## Desired behavior

When I install Solidus 3.2 with the payment method set to the default (PayPal)
Then I should be able to utilize the payment method without additional manual integration

## Current behavior

After you install Solidus 3.2 with the payment method set to PayPal, you would still need to update the SolidusPaypalCommercePlatform gem from `0.6` to `~> 1.0.0.beta2` and then rerun the `rails generate solidus_paypal_commerce_platform:install`. For example, please check out this [Rails app](https://github.com/gsmendoza/solidus_store/tree/rails-7-0-4--solidus-3-2-4--solidus-paypal-commerce-platform) to see how I set it up to work with PayPal. The Rails app was used to test https://github.com/solidusio/solidus_paypal_commerce_platform/pull/173.

## Analysis

I think the version of SolidusPaypalCommercePlatform that is installed is `0.6` because the latest version is still in beta (1.0.0.beta2).

During Solidus installation, we do warn the user that "selecting a payment along with `solidus_starter_frontend` might require manual integration" (https://github.com/solidusio/solidus/blob/v3.2/core/lib/generators/solidus/install/install_generator.rb#L143). But unfortunately, we don't have documentation right now on how to do that with PayPal.

## Solution

Fortunately, with SolidusPaypalCommercePlatform 1.0.0.beta2, PayPal integration now works out of the box with `solidus_starter_frontend`.

## Manual testing

1. Check out the branch.
2. Generate the sandbox (`bin/sandbox`)
3. With the sandbox server running, set up the PayPal payment method.
3. You should be able to do a checkout via PayPal.

For integrating and testing PayPal as a payment method, you can follow the instructions at https://github.com/solidusio/solidus_paypal_commerce_platform/pull/177. It's still a work in progress.

## Screenshots

![image](https://user-images.githubusercontent.com/61476/208369241-77a8ce02-01b9-4371-ba81-f086ba35d0a6.png)

![image](https://user-images.githubusercontent.com/61476/208369317-f23a136c-50ca-4070-9d74-98a42bd2ee96.png)


## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [x] I have attached screenshots to demo visual changes.
- [x] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [x] I have updated the README to account for my changes.
